app/services/samples/clone_service.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Samples
+  # Service used to clone samples
+  class CloneService < BaseProjectService
+    CloneError = Class.new(StandardError)
+
+    def execute(new_project_id, sample_ids)
+      validate(new_project_id, sample_ids)
+
+      authorize! @project, to: :clone_sample?
+
+      @new_project = Project.find_by(id: new_project_id)
+      authorize! @new_project, to: :clone_sample_into_project?
+
+      clone_samples(sample_ids)
+    rescue Samples::CloneService::CloneError => e
+      @project.errors.add(:base, e.message)
+      {}
+    end
+
+    private
+
+    def validate(new_project_id, sample_ids)
+      raise CloneError, I18n.t('services.samples.clone.empty_new_project_id') if new_project_id.blank?
+
+      raise CloneError, I18n.t('services.samples.clone.empty_sample_ids') if sample_ids.blank?
+
+      return unless @project.id == new_project_id.to_i
+
+      raise CloneError, I18n.t('services.samples.clone.same_project')
+    end
+
+    def clone_samples(sample_ids)
+      cloned_sample_ids = {}
+      sample_ids.each do |sample_id|
+        cloned_sample_ids[sample_id] = clone_sample(sample_id)
+      rescue ActiveRecord::RecordInvalid
+        @project.errors.add(:sample, I18n.t('services.samples.clone.sample_exists', sample_id:))
+      end
+      cloned_sample_ids
+    end
+
+    def clone_sample(sample_id)
+      sample = Sample.find_by(id: sample_id, project_id: @project.id)
+      clone = Sample.new(name: sample.name, description: sample.description, project_id: @new_project.id)
+      clone_metadata(sample, clone) if clone.valid?
+      clone_attachments(sample, clone) if clone.valid?
+      clone.save! if @project.errors.empty?
+      clone.id
+    end
+
+    def clone_metadata(sample, clone)
+      metadata_changes = Samples::Metadata::UpdateService.new(@new_project, clone, current_user,
```

Comment:
Talked to @deepsidhu85 about this and realized that calling the `Metadata::UpdateService` does not work well as the cloned samples will technically have 'new' metadata, the provider will be overwritten by the user who is cloning the sample, and the `metadata_provenance` will all be changed and updated as well.

The best way to handle this would be simply doing something along the lines of `clone.metadata = original_sample.metadata` and `clone.metadata_provenance = original_sample.metadata_provenance`. And then updating the `metadata_summary` of the project and its parents by creating an `update_metadata_summary_by_sample_addition` function after the clones have been validated.

If you look at `services/samples/destroy_service`, follow along the `sample.project.namespace.update_metadata_summary_by_sample_deletion(sample)` call, which will have similar behaviour as what you need, you would just add to `metadata_summary` rather than subtract.

Reply:
Maybe a `Metadata::CreateService` for `new` metadata?

Reply:
@ericenns What are your thoughts?

Reply:
c91470a611e0cff46939792fa6fd83a8fc863641